### PR TITLE
Fix scm and package update

### PR DESCRIPTION
**Pull request by maheshp**
Issue: #8542 

Description: Update pipeline config on changes to to SCM, Package repo or package.

### CODE REVIEW

File: server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageConfigCommand.java [hunk 1]
```diff
@@ -68,6 +75,17 @@ public boolean canContinue(CruiseConfig cruiseConfig) {
         return isAuthorized() && isRepositoryPresent(cruiseConfig) && isIdSame() && isRequestFresh();
     }
 
+    private void updatePackageConfigurationOnAssociatedPipelines(CruiseConfig modifiedConfig) {
+        List<PipelineConfig> pipelinesWithPackage = modifiedConfig.pipelinesAssociatedWithPackage(newPackage);
+        pipelinesWithPackage.forEach(pipelineConfig -> {
+            pipelineConfig.packageMaterialConfigs().forEach(packageMaterialConfig -> {
+                if (packageMaterialConfig.getPackageId().equals(newPackage.getId())) {
```
Comment (kritika-singh3) [inline server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageConfigCommand.java:82]: The comparison check has been performed in the `modifiedConfig.pipelinesAssociatedWithPackage` method. Do we require it again.

File: server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageRepositoryCommand.java [hunk 2]
```diff
@@ -50,16 +53,29 @@ public void update(CruiseConfig modifiedConfig) {
...
     @Override
     public boolean canContinue(CruiseConfig cruiseConfig) {
         return super.canContinue(cruiseConfig) && isIdSame() && isRequestFresh();
     }
 
+    private void updatePackageRepositoryConfigurationOnAssociatedPipelines(CruiseConfig modifiedConfig) {
+        List<PipelineConfig> pipelinesWithPackageRepo = modifiedConfig.pipelinesAssociatedWithPackageRepository(newRepo);
+        pipelinesWithPackageRepo.forEach(pipelineConfig -> {
+            pipelineConfig.packageMaterialConfigs().forEach(packageMaterialConfig -> {
+                if (packageMaterialConfig.getPackageDefinition().getRepository().getId().equals(newRepo.getId())) {
```
Comment (kritika-singh3) [inline server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageRepositoryCommand.java:69]: Same here: the check has performed in `modifiedConfig.pipelinesAssociatedWithPackageRepository`

File: server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageConfigCommand.java (hunk 1 above)
Comment (maheshp) [reply server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageConfigCommand.java:82]: I was thinking of a scenario where a PipleineConfig has more than one package material. We would need this check in those cases right?

Comment (kritika-singh3) [reply server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageConfigCommand.java:82]: Yes right - we would need this check then.

**Review by kritika-singh3 — APPROVED** [review]
LGTM. Neither of the 2 scenarios mentioned in #8542 came up while testing.

**Comment by kritika-singh3** [comment]
Verified on `GoCD Version: 20.9.0 (12264-30bf28b07277798dfc36761dad864e170cee5336)`

**Pull request merged**